Added type definitions for @commerce7/admin-ui

Weldawadyathink opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

DangerBotOSS commented:
<!--
  0 failure: 
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->



## Formatting

The following files are not formatted:
1. types/commerce7__admin-ui/commerce7__admin-ui-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7f7777f71fbc33e23e9010a68e3954ff49f48304
</p>

peterblazejewicz commented:
you would have to answer typical questions:
- why you have those exclusion in ESLint configuration file (could those be at least used inline)
- was this declaration autogenerated? Hence the need for ESLint exclusions

Weldawadyathink commented:
> you would have to answer typical questions:
> 
> - why you have those exclusion in ESLint configuration file (could those be at least used inline)
> 
> - was this declaration autogenerated? Hence the need for ESLint exclusions

It isn't auto generated, I just didn't think about putting the eslint config inline. The only eslint I needed to override was only having one declare module statement. Let me know if there is anything else that needs a second pass, and I can update the PR.

rbuckton commented:
That's a fair point. The less you need to override in the eslint configuration, the better.

rbuckton commented on types/commerce7__admin-ui/.eslintrc.json:
```diff
@@ -0,0 +1,11 @@
+{
+    "rules": {
+        "@definitelytyped/no-any-union": "off",
```
Per @peterblazejewicz, handling these inline or adjusting the types to not require disabling the lint rules would be preferred.

Weldawadyathink commented:
Updated to (hopefully) address all requests. The eslint configs are gone. After some discussion on the discord, I found out I don't need the `declare module`, so I changed it to export the component classes directly. Please let me know if there are any other issues.

jakebailey commented on types/commerce7__admin-ui/commerce7__admin-ui-tests.ts:
```diff
@@ -0,0 +1 @@
+import { Commerce7AdminUI } from "@commerce7/admin-ui";
```
This seems like not very many tests; should this be a `.tsx` file and then actually use the components?

Weldawadyathink replied:
Yeah, I will admit I didn't really know what tests to write. Do you know of a react component library that is similar that I could use for some guidance? I looked through the typings for material-ui, but that library seems so different than @commerce7/admin-ui that the tests didn't look relevant.

jakebailey replied:
I don't have any specific examples, but if I were looking I would just be finding another package with a `.tsx` file.

Weldawadyathink replied:
Good idea, I will try that once I have some more time to work on this. Thanks for the nudge in the right direction.

Weldawadyathink commented:
Added some testing. For the most part, its a copy-paste of the example code from the library's [sample code](https://admin-ui-docs.commerce7.com/). I ended up finding a few errors on my end, and a few component properties that are not documented correctly in their docs, at least according to their examples.

Weldawadyathink commented:
Ready to merge
